tests/test_data.py

```diff
@@ -189,6 +189,25 @@ def test_move_contents(self):
         self.assertTrue(os.path.isfile(os.path.join(baz, "spam.txt")))
         self.assertTrue(os.path.isdir(os.path.join(baz, "fizz")))
 
+        # create a copy of the file and directory currently in baz in foo
+        if relentless.mpi.world.rank_is_root:
+            open(dfoo.file("spam.txt"), "w").close()
+        dfoo.directory("fizz", create=relentless.mpi.world.rank_is_root)
+        relentless.mpi.world.barrier()
+        self.assertTrue(os.path.isfile(os.path.join(foo, "spam.txt")))
+        self.assertTrue(os.path.isdir(os.path.join(foo, "fizz")))
+        self.assertTrue(os.path.isfile(os.path.join(baz, "spam.txt")))
+        self.assertTrue(os.path.isdir(os.path.join(baz, "fizz")))
+
+        # move file and directory from foo to baz
+        if relentless.mpi.world.rank_is_root:
+            dfoo.move_contents(baz)
+        relentless.mpi.world.barrier()
+        self.assertFalse(os.path.isfile(os.path.join(foo, "spam.txt")))
+        self.assertFalse(os.path.isdir(os.path.join(foo, "fizz")))
+        self.assertTrue(os.path.isfile(os.path.join(baz, "spam.txt")))
+        self.assertTrue(os.path.isdir(os.path.join(baz, "fizz")))
+
```

Comment:
There should be a short unit test added for the `overwrite` option of the optimizers in that file.